ui/src/plot.tsx

```diff
@@ -1129,7 +1129,7 @@ export const
                     if (e.stateStatus && e.state === 'selected' && model.name && e.element.geometry.customOption.interactive) {
                       const ret = Array.isArray(e.element?.data)
                         ? e.element.data.map(({ idx }: any) => ({ idx, ...originalDataRef.current[idx] }))
```

Comment:
You probably forgot to update `idx` to `wave_idx` when handling events in case you got array of objects.

```suggestion
                        ? e.element.data.map(({ wave_idx }: any) => ({ idx: wave_idx, ...originalDataRef.current[wave_idx] }))
```
Property `idx` does not exist when clicking on line:

<img width="1405" alt="image" src="https://github.com/h2oai/wave/assets/23740173/79dbea9e-7c61-4352-9d76-925bb6c3d314">

Or another solution could be:
```suggestion
                        ?  e.element.data.map((_item: any, idx: I) => ({ idx, ...originalDataRef.current[idx] }))
```

Reply:
Thanks for catching that.

> Or another solution could be:

I went for the first suggestion as this one relies on loop indices instead the object ones, which would be a problem in case the data is sorted in a different order than arrived.